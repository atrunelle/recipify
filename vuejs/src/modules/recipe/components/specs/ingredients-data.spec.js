import { mount } from '@vue/test-utils';

import IngredientsDat from '../ingredients-data';

describe('Component: Ingredients data', () => {
  let component;

  beforeEach(() => {
    const totalNutrients = [{
      totalQuantity: {
        label: 'Protein',
        quantity: 100,
        percentage: 50,
        unit: 'gr',
      },
      totalDaily: {
        label: 'Protein',
        quantity: 20,
        unit: '%',
      },
    }, {
      totalQuantity: {
        label: 'Fat',
        quantity: 100,
        percentage: 10,
        unit: 'gr',
      },
      totalDaily: {
        label: 'Fat',
        quantity: 40,
        unit: '%',
      },
    }];

    component = mount(IngredientsDat, {
      stubs: ['nutrients-macro-chart'],
      propsData: {
        totalNutrients,
        totalCalories: 120,
        totalWeight: 100,
      },
    });
  });

  it('should match snapshot', () => {
    expect(component.html()).toMatchSnapshot();
  });
});
