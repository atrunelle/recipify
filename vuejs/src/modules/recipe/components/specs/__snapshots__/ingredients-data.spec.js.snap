// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Ingredients data should match snapshot 1`] = `
<div class="flex xs12">
  <div class="v-card v-sheet theme--light">
    <div class="v-card__title v-card__title--primary">
      <h2 class="headline">Total nutrients</h2>
    </div>
    <div class="v-card__text">
      <h3 class="subheading">Calories: 120cal</h3>
      <h3 class="subheading">Total Weight: 100gr</h3>
      <div class="layout align-end justify-space-between">
        <div class="flex text-xs-center">
          <p class="body-1">Total Nutrients</p>
          <p class="body-1">Total Daily percentage</p>
        </div>
        <div class="flex text-xs-center">
          <h3 class="subheading pa-2"></h3>
          <p class="body-1">100.00gr</p>
          <p class="body-1">20.00%</p>
        </div>
        <div class="flex text-xs-center">
          <h3 class="subheading pa-2"></h3>
          <p class="body-1">100.00gr</p>
          <p class="body-1">40.00%</p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
